staging/src/k8s.io/api/networking/v1/types.go

```diff
@@ -49,7 +49,11 @@ const (
 	PolicyTypeEgress PolicyType = "Egress"
 )
 
-// NetworkPolicySpec provides the specification of a NetworkPolicy
+// NetworkPolicySpec provides the specification of a NetworkPolicy.  A suite of network
+// policies allow traffic ONLY if BOTH the egress AND the ingress are allowed.  For example,
```

Comment:
I'm not sure this belongs on the NeworkPolicySpec definition. What we're really talking about here is more best-practice of how to deploy multiple policies across multiple difference services, which isn't really part of the NetworkPolicy API, as I see it.

I feel like this needs to be documented, but might be better placed as a note in the user documentation rather than in the API definition?

Reply:
A PR against the documentation was merged yesterday: https://github.com/kubernetes/website/pull/25965 

I still believe adding this documentation in the spec itself makes sense to ensure that CNI providers implementing the spec are aware of the egress/ingress combination case.

Reply:
Has there been any evidence of a network policy implementer getting this wrong? I feel like this can be derived from the definition of PodSelector / Ingress / Egress, no?

> ingress is allowed but egress is not, or egress is allowed but ingress is not, the traffic will not be allowed

Either way, the way it is worded here is not right. It implies that Ingress and Egress need to be set on the same NP in order for traffic to be allowed, which isn't accurate.

If anything, we might want to say "In order for a connection to succeed, it must be allowed by any egress rules applied to the source pods as well as any ingress rules applied to the destination pods".

Reply:
Happy to fix up the wording if you do support adding something here @caseydavenport , also fine with closing this PR unmerged if the right place to have this is in Nills' PR -- I could go either way on this 😄 !

My main motivation for adding this is I think a lot of people -- myself included -- get tripped up on this case, even after reading through types.go several times I had no idea that it would work this way.

Reply:
Yes, there is a case where a CNI provider (Azure CNI) implemented this the wrong way, documented here: https://github.com/Azure/AKS/issues/1158 . 
It's that implementation that led us to create the test case and add this documentation to clarify.

Reply:
@caseydavenport do you think this is worth moving forward with in some form, or should we just close it?  I'm fine either way!

Reply:
Sorry for the delay -

I'm OK with something simple like this:

> In order for a connection to succeed, it must be allowed by any egress rules applied to the source pods as well as any ingress rules applied to the destination pods"

But like I hinted at above, if this isn't apparent or derivable from the definition of ingress / egress rules then that is probably where we should fix things.

Reply:
👍 updated to make that change, thanks @caseydavenport !!